File: backend/microservice/src/admin/director/objects/DirectorPermissionSummary.scala
package microservice.admin.director.objects

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}
import microservice.system.objects.AdminLevel

final case class DirectorPermissionSummary(
  userId: String,
  adminLevel: AdminLevel,
  canManageUiCustomization: Boolean
)

object DirectorPermissionSummary {
  implicit val encoder: Encoder[DirectorPermissionSummary] = deriveEncoder
  implicit val decoder: Decoder[DirectorPermissionSummary] = deriveDecoder
}

File: frontend/src/objects/admin/director/permissions/director-permission-summary.ts
import { z } from "zod";
import { AdminLevelSchema } from "../../../system/system-objects.js";

export const DirectorPermissionSummarySchema = z.object({
  userId: z.string().min(1),
  adminLevel: AdminLevelSchema,
  canManageUiCustomization: z.boolean(),
});

export type DirectorPermissionSummary = z.infer<typeof DirectorPermissionSummarySchema>;
